### PR TITLE
Support tabDragStart/tabDragStop events.

### PR DESCRIPTION
**Pull request by lincolnthree**
Addresses https://github.com/zyra/ionic2-super-tabs/issues/347 and a few prior, now-closed requests.

### CODE REVIEW

File: src/components/super-tabs.ts [hunk 1]
```diff
@@ -204,6 +216,19 @@ export class SuperTabsComponent
 
   private maxIndicatorPosition: number;
 
+  /**
+   * The duration (ms) after the last onDrag event where the
+   * dragging state is considered 'stopped'
+   */
+  private tabDragStoppedTimeout: number = 200;
```
Comment (ihadeed) [inline src/components/super-tabs.ts]: this should be part of the config so users can customise it

File: src/components/super-tabs.ts [hunk 2]
```diff
@@ -204,6 +216,19 @@ export class SuperTabsComponent
...
+
+  /**
+   * We use a timeout to identify whether dragging has completed
+   * @type {number}
+   * @private
+   */
+  private _isDraggingTimeoutId: number | null = null;
```
Comment (ihadeed) [inline src/components/super-tabs.ts]: Instead of using/defining it as null, it should be undefined.

File: src/components/super-tabs.ts [hunk 3]
```diff
@@ -755,6 +783,26 @@ export class SuperTabsComponent
...
+    this._isDraggingTimeoutId = setTimeout(() => {
+      this.tabDragStopped();
+    }, this.tabDragStoppedTimeout);
+  }
+
+  private tabDragStopped() {
+    // we have no mechanism to identify when the user stops dragging
+    // so we need to use a timeout which is reset on every drag motion
+    this._isDraggingTimeoutId && clearTimeout(this._isDraggingTimeoutId);
+    this._isDraggingTimeoutId = null;
```
Comment (ihadeed) [inline src/components/super-tabs.ts]: `null` > `undefined`

**Comment by lincolnthree** [comment]
I agree with "We can know exactly when the dragging stopped if we listen to the container events & push it up to the parent."

Actually I think it is necessary as the current approach using timeouts actually kills performance pretty horribly since it's constantly checking and cancelling the timeout. I took this PR originally from one you got a while ago, I'll update with the changes, but... I'm not sure what the best way to push events from container up is. It looks like I'll need to add a hook to the parent in the container, or vis-versa.

Regarding your specific code change requests/comments. I think these are all rendered non-applicable if we change this to use the actual container events, right?

**Comment by lincolnthree** [comment]
Alright. Take a look at this. I think it's pretty clean now. Much more efficient for sure.

I'm not sure of the nuances of your gesture events, so hopefully I put things in the right place. Let me know what you think!

**Comment by lincolnthree** [comment]
hey @ihadeed. Just checking on your thoughts about this PR since update. I have a few more things I might send your way as well if I can get them cleaned up (including a "low graphics" mode") and the ability to pass through lifecycle to the active tab (willEnter, didEnter, willLeave, didLeave) etc.

**Pull request merged**